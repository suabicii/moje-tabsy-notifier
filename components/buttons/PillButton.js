import React from "react";
import {TouchableOpacity} from "react-native";
import {Text} from "react-native";

function PillButton({handlePress, loading, variant, text}) {
    let backgroundColor;
    if (variant === 'primary') {
        backgroundColor = '#78c2ad';
    } else if (variant === 'warning') {
        backgroundColor = '#f38c4c';
    } else {
        backgroundColor = '#525252';
    }
    
    return (
        <TouchableOpacity
            testID="btn-pill"
            onPress={handlePress}
            style={{
                backgroundColor,
                borderRadius: 24,
                height: 50,
                justifyContent: "center",
                marginBottom: 20,
                marginHorizontal: 15
            }}
        >
            <Text style={{
                color: "#fff",
                fontSize: 24,
                fontWeight: "300",
                textAlign: "center"
            }}>
                {loading ? 'Proszę czekać...' : text}
            </Text>
        </TouchableOpacity>
    );
}

export default PillButton;