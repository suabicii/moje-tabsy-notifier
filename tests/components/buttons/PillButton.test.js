import React from "react";
import renderer from "react-test-renderer";
import PillButton from "../../../components/buttons/PillButton";

it('should correctly render PillButton', () => {
    const handlePress = jest.fn();
    const tree = renderer.create(<PillButton handlePress={handlePress}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('should change text value when loading is true', () => {
    const handlePress = jest.fn();
    const btnBeforeLoading = renderer.create(
        <PillButton handlePress={handlePress}/>
    ).toJSON();
    const btnAfterLoading = renderer.create(
        <PillButton handlePress={handlePress} loading={true}/>
    ).toJSON();
    expect(btnBeforeLoading.children[0].children)
        .not
        .toEqual(btnAfterLoading.children[0].children);

    //EXPLANATION:
    /*
       console.log(btnBeforeLoading.children);
    * [
        {
          type: 'Text',
          props: { style: [Object] },
          children: [ 'Zaloguj siÄ™' ]
        }
    ]
    * */
});