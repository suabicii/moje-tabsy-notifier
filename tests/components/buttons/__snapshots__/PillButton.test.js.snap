// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render PillButton 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#525252",
      "borderRadius": 24,
      "height": 50,
      "justifyContent": "center",
      "marginBottom": 20,
      "marginHorizontal": 15,
      "opacity": 1,
    }
  }
  testID="btn-pill"
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 24,
        "fontWeight": "300",
        "textAlign": "center",
      }
    }
  />
</View>
`;
